lib/job.js

```diff
@@ -131,6 +131,28 @@ class Job extends BaseModel {
 
         return factory.list(listConfig);
     }
+
+    /**
+     * Remove all builds associated with this job and the job itself
+     * @return {Promise}        Resolves to null if remove successfully
+     */
+    remove() {
+        const removeBuilds = (() =>
+            // Iterate through the builds and remove them
+            this.getBuilds().then(builds => {
+                if (builds.length > 0) {
+                    builds.forEach(build => Promise.all(build.remove()));
```

Comment:
What concerns you in particular? This would only resolves when all the builds are removed. Do you have any suggestions? 

@petey  Any better way to write this?